apps/openmw/mwworld/projectilemanager.cpp

```diff
@@ -92,7 +94,18 @@ namespace MWWorld
             attachTo = rotateNode;
         }
 
-        mResourceSystem->getSceneManager()->getInstance(model, attachTo);
+        osg::ref_ptr<osg::Node> ptr = mResourceSystem->getSceneManager()->getInstance(model, attachTo);
+
+        if (state.mIdMagic.size() > 1)
+            for (size_t iter = 1; iter != state.mIdMagic.size(); ++iter)
+            {
+                char numstr[8];
+                sprintf(numstr, "%d", (int)iter);
+                std::string node = "Dummy0";
+                node = node + numstr;
+                const ESM::Weapon* weapon = MWBase::Environment::get().getWorld()->getStore().get<ESM::Weapon>().find (state.mIdMagic.at(iter));
+                mResourceSystem->getSceneManager()->getInstance("meshes\\" + weapon->mModel, attachTo);
```

Comment:
The missing piece here is that you want to attach to the DummyX node instead of attachTo.

To find a node by name use the SceneUtil::FindByNameVisitor.

<pre>
SceneUtil::FindByNameVisitor findVisitor(node);
attachTo->accept(findVisitor);
// findVisitor.mFoundNode now holds a pointer to the node, or NULL if not found 
</pre>